.github/workflows/ci.yml

```
@@ -112,7 +139,8 @@ jobs:
           retention-days: 1
 
   sonarcloud:
-    needs: [build-linux, build-ios]
+    needs: [changes, build-linux, build-ios]
+    if: needs.changes.outputs.source == 'true'
     runs-on: ubuntu-latest
     if: github.actor != 'dependabot[bot]'
```

actionlint (1.7.10)

[error] key "if" is duplicated in "sonarcloud" job. previously defined at line:143,col:5

(syntax-check)